src/mpid/ch4/src/mpidig_recv_utils.h

```diff
@@ -46,6 +50,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_recv_check_rndv_cb(MPIR_Request * rreq)
         if (MPIDIG_REQUEST(rreq, req)) {
             MPIDIG_REQUEST(rreq, req->recv_async).data_copy_cb = NULL;
         }
+        MPIR_Request_free_with_safety(rreq, false);
```

Comment:
At least today we seem to avoid using `MPIR_Request_free_with_safety` directly in code.
```suggestion
        MPIR_Request_free_unsafe(rreq);
```